### PR TITLE
feat_: accounts/saved addresses capacity related api endpoints added

### PR DESCRIPTION
**Pull request by saledjenic**
Added endpoints:
- `RemainingAccountCapacity`
- `RemainingKeypairCapacity`
- `RemainingWatchOnlyAccountCapacity`
- `RmaininngCapacityForSavedAddresses`

Backend part of https://github.com/status-im/status-desktop/issues/15934

### CODE REVIEW

**Comment by saledjenic** [comment]
@ilmotta @briansztamfater not sure who is going to implement these limitations on the mobile app side, but just to be aware of these new endpoints.

File: services/ext/api.go [hunk 1]
```diff
@@ -1126,6 +1126,14 @@ func (api *PublicAPI) GetSavedAddresses(ctx context.Context) ([]*wallet.SavedAdd
 	return api.service.messenger.GetSavedAddresses(ctx)
 }
 
+func (api *PublicAPI) GetSavedAddressesPerMode(ctx context.Context, testnetMode bool) ([]*wallet.SavedAddress, error) {
+	return api.service.messenger.GetSavedAddressesPerMode(testnetMode)
+}
+
+func (api *PublicAPI) RmaininngCapacityForSavedAddresses(ctx context.Context, testnetMode bool) (int, error) {
```
Comment (ilmotta) [inline services/ext/api.go]: Typo

File: services/wallet/saved_addresses.go [hunk 2]
```diff
@@ -115,11 +117,28 @@ func (sam *SavedAddressesManager) GetSavedAddresses() ([]*SavedAddress, error) {
...
+func (sam *SavedAddressesManager) GetSavedAddressesPerMode(testnetMode bool) ([]*SavedAddress, error) {
+	return sam.getSavedAddresses(fmt.Sprintf("is_test = %t AND removed != 1", testnetMode))
+}
+
 // GetRawSavedAddresses provides access to the soft-delete and sync metadata
 func (sam *SavedAddressesManager) GetRawSavedAddresses() ([]*SavedAddress, error) {
 	return sam.getSavedAddresses("")
 }
 
+func (sam *SavedAddressesManager) RmaininngCapacityForSavedAddresses(testnetMode bool) (int, error) {
```
Comment (ilmotta) [inline services/wallet/saved_addresses.go]: Typo

File: services/accounts/accounts.go [hunk 3]
```diff
@@ -211,6 +211,18 @@ func (api *API) AddKeypair(ctx context.Context, password string, keypair *accoun
 	return nil
 }
 
+func (api *API) RemainingAccountCapacity(ctx context.Context) (int, error) {
+	return (*api.messenger).RemainingAccountCapacity()
+}
+
+func (api *API) RemainingKeypairCapacity(ctx context.Context) (int, error) {
```
Comment (ilmotta) [inline services/accounts/accounts.go:220]: We may want to more diligently document our public API. The name of the endpoints are usually self-explanatory, but they don't give any recommendation about when to use, why, potential concerns, etc. Could be really helpful for a developer without context.

Some level of documentation on the public API could be our minimum standard in the future, something for the status-go council to think about :) cc @igor-sirotin @Samyoul 

If you think there's nothing too interesting to document it's okay 👍🏼

**Comment by saledjenic** [comment]
@ilmotta thanks, updated.

**Review by xjc90s — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Pull request merged**